Lib9c.Policy/NCStagePolicy.cs

```diff
@@ -61,10 +63,10 @@ public IEnumerable<Transaction> Iterate(BlockChain blockChain, bool filtered = t
                     s.Add(tx);
                     int txQuotaPerSigner = _quotaPerSigner;
 
-                    // update txQuotaPerSigner if ACS returns a value for the signer.
-                    if (_accessControlService?.GetTxQuota(tx.Signer) is { } acsTxQuota)
+                    // update txQuotaPerSigner if signer is in the list
+                    if (_quotaPerSignerList.TryGetValue(tx.Signer, out int signerQuota))
```

Comment:
Ah I understand, I think `SetItem` is right choice then.

Just to be clear, are you referring to https://learn.microsoft.com/ko-kr/dotnet/api/system.collections.immutable.immutabledictionary-2.setitem?view=net-8.0#system-collections-immutable-immutabledictionary-2-setitem(-0-1)?

Reply:
Do you think the `AddOrUpdate` method in `ConcurrentDictionary` work here as well?

https://learn.microsoft.com/ko-kr/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-8.0#system-collections-concurrent-concurrentdictionary-2-addorupdate(-0-system-func((-0-1))-system-func((-0-1-1)))

Reply:
> Just to be clear, are you referring to [learn.microsoft.com/ko-kr/dotnet/api/system.collections.immutable.immutabledictionary-2.setitem?view=net-8.0#system-collections-immutable-immutabledictionary-2-setitem(-0-1)](https://learn.microsoft.com/ko-kr/dotnet/api/system.collections.immutable.immutabledictionary-2.setitem?view=net-8.0#system-collections-immutable-immutabledictionary-2-setitem(-0-1))?

Yes, but I mean `Dictionary.SetItem` because `_quotaPerSignerList` is `Dictionary<K,V>`.

```csharp
private readonly Dictionary<Address, int> _quotaPerSignerList;
```

> Do you think the AddOrUpdate method in ConcurrentDictionary work here as well?
>
> [learn.microsoft.com/ko-kr/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-8.0#system-collections-concurrent-concurrentdictionary-2-addorupdate(-0-system-func((-0-1))-system-func((-0-1-1)))](https://learn.microsoft.com/ko-kr/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=net-8.0#system-collections-concurrent-concurrentdictionary-2-addorupdate(-0-system-func((-0-1))-system-func((-0-1-1))))

Well. I didn't understand `ConcurrentDictionary.AddOrUpdate` well, but the `updateValueFactory` parameter appears to create a new value based on an existing value. But since we're fetching from Redis, I'm not sure if this is necessary. Using a ConcurrentDictionary might be a good idea since it can be accessed by multiple threads, and tasks, but I'm not sure about using AddOrUpdate, or simply ReadWriteLock might be a good idea. You can access it a lot when it's a ReadLock. I'm not sure how well ConcurrentDictionary supports this, so I won't add more.

Reply:
> Yes, but I mean Dictionary.SetItem because _quotaPerSignerList is Dictionary<K,V>.

![image](https://github.com/planetarium/lib9c/assets/42176649/bbcc01d9-16a7-4d73-817b-0ad2b05f94c4)

I also thought you meant `Dictionary.SetItem` but I don't think there's a `SetItem` method in `Dictionary<K,V>` as indicated in the snapshot. Maybe it's new in `.NET8`? 🤔 

If that method isn't available in `.NET6`, then this can be an alternative 😇:

```
                if (_quotaPerSignerList.ContainsKey(transaction.Signer))
                {
                    _quotaPerSignerList[transaction.Signer] = acsTxQuota;
                }
                else
                {
                    _quotaPerSignerList.TryAdd(transaction.Signer, acsTxQuota);
                }

```

Reply:
Ah, maybe it can be like the below example. (I missed it 😥)

```
_quotaPerSignerList[transaction.Signer] = acsTxQuota;
```

Ref: https://learn.microsoft.com/ko-kr/dotnet/api/system.collections.generic.idictionary-2.item?view=net-8.0#system-collections-generic-idictionary-2-item(-0)

Reply:
Ah ok, thanks for checking! I'll update the PR accordingly 🙏